packages/superset-ui-core/src/query/types/Column.ts

```diff
@@ -55,3 +56,15 @@ export default {};
 export function isPhysicalColumn(column: AdhocColumn | PhysicalColumn): column is PhysicalColumn {
   return typeof column === 'string';
 }
+
+export function isColumnMeta(column: AdhocColumn | ColumnMeta): column is ColumnMeta {
+  return 'column_name' in column;
+}
+
+export function isSavedExpression(column: AdhocColumn | ColumnMeta): column is ColumnMeta {
+  return 'column_name' in column && 'expression' in column;
+}
+
+export function isAdhocColumn(column: AdhocColumn | ColumnMeta): column is AdhocColumn {
+  return 'label' in column && 'sqlExpression' in column;
+}
```

Comment:
I think we need to move these type guards to `superset-ui/chart-controls` as `superset-ui/core` doesn't have a dependency on `superset-ui/chart-controls`